packages/common/lib/xml-ns-key.js

```diff
@@ -0,0 +1,38 @@
+// The xml parser takes care of validating the attribute name.
+// If the user started the attribute name with "xmlns:" we can assume that
+// they meant for it to be an xml namespace attribute.
+// xmlns attributes explicitly can't contain ":" after the "xmlns:" part.
+const namespaceRegex = /^xmlns(?<prefixWithColon>:(?<prefix>[^:]*))?$/;
+
+function isXMLNamespaceKey(key, includeEmptyPrefix) {
+  if (typeof key !== "string") {
+    return false;
+  }
+  const matchArr = key.match(namespaceRegex);
+  if (matchArr === null) {
+    return false;
+  }
+  if (includeEmptyPrefix === true) {
+    return true;
+  }
+  /* istanbul ignore next - defensive coding */
+  const groups = matchArr.groups || {};
+  // In the case where key === "xmlns:", prefix will be empty and prefixWithColon will not be empty
+  return !(groups.prefixWithColon && !groups.prefix);
```

Comment:
Would it be simpler to have more distinct groups in the regExp?
1. xmlns
2. colon
3. prefix

and then ask if 1 &2 exists (empty prefix case) or 1 & 2 & 3 exists (proper xmlns)

Reply:
I changed the condition a bit to make it clearer, let me know if you think I should still change this